crates/ruma-client-api/src/membership/mutual_rooms.rs

```diff
@@ -0,0 +1,48 @@
+//! `GET /_matrix/client/*/user/mutual_rooms/{user_id}`
+
+pub mod unstable {
+    //! `/unstable/` ([spec])
+    //!
+    //! [spec]: https://github.com/matrix-org/matrix-spec-proposals/blob/hs/shared-rooms/proposals/2666-get-rooms-in-common.md
+
+    use ruma_common::{api::ruma_api, OwnedRoomId, UserId};
+
+    ruma_api! {
+        metadata: {
+            description: "Get mutual rooms with another user.",
+            method: GET,
+            name: "mutual_rooms",
+            unstable_path: "/_matrix/client/unstable/uk.half-shot.msc2666/user/mutual_rooms/:user_id",
+            rate_limited: true,
+            authentication: AccessToken,
+        }
+
+        request: {
+            /// The user to search mutual rooms for.
+            #[ruma_api(path)]
+            pub user_id: &'a UserId,
+        }
+
+        #[derive(Default)]
```

Comment:
I'm not sure there's any value in this trait impl, mind removing it?

Reply:
oops, sorry, copy and paste error